### PR TITLE
[styles] Add sport beachvolleyball, chess, skateboard, table_tennis, volleyball, 9pin, 10pin, climbing

### PR DESCRIPTION
**Pull request by j13m126**
grafik
grafik
icon sources:
[volleyball&beachvolleyball](https://fonts.google.com/icons?icon.set=Material+Icons&icon.query=volleyball)
[chess](https://fontawesome.com/icons/chess?s=solid)
[skateboarding](https://fonts.google.com/icons?icon.set=Material+Icons&icon.query=skateboard)
[table_tennis](https://fontawesome.com/icons/table-tennis-paddle-ball?s=solid)

### CODE REVIEW

**Comment by biodranik** [comment]
Thank you for the pull request!

1. All icon sources are already listed in our licenses, right?
2. Icons should be readable on most backgrounds and also look good with the dark style. Current round ones were designed to look good in all cases. But we want to focus on the rule "round background means a business of some sort" and use lighter icons instead. That may require a white-line-with-dark-outline or dark-line-with white-outline style.
3. It would be better to have the same style for all icons. Can you please try to remove circles for existing sports icons in a separate commit (and maybe edit them a bit for better visibility)?

**Comment by vng** [comment]
Better to have only two last commits: 
- [string] Regenerated
- [styles] Regenerated
per PR.
And make rebase to avoid conflicts before regenerating.

**Comment by biodranik** [comment]
The best way to avoid rebase conflicts is to remove regenerated commits and then regenerate them again.

File: data/categories.txt [hunk 1]
```diff
@@ -6801,6 +6950,117 @@ vi:Quần vợt
...
+sv:Kägel
+sw:Wkittles
+th:Skittles
+tr:Skittles
+uk:Кеглі
+vi:Nhảy nhót
+zh-Hans:九柱戏
+zh-Hant:九柱戲
+
+sport.10pin
```
Comment (biodranik) [inline data/categories.txt:7027]:
```suggestion
sport-10pin
```

File: data/categories.txt [hunk 2]
```diff
@@ -6801,6 +6950,117 @@ vi:Quần vợt
...
+sv:Volleyboll
+sw:Voliboli
+th:วอลเลย์บอล
+tr:Voleybol
+uk:Волейбол
+vi:Bóng chuyền
+zh-Hans:排球
+zh-Hant:排球
+
+sport=9pin
```
Comment (biodranik) [inline data/categories.txt:6990]:
```suggestion
sport-9pin
```

File: data/categories.txt [hunk 3]
```diff
@@ -6801,6 +6950,117 @@ vi:Quần vợt
...
+en:Nine-pin Bowling
```
Comment (biodranik) [inline data/categories.txt:6991]: Does it make sense to differentiate these two types? Let's merge them into one to avoid copy-paste. CC @vng

**Review by biodranik — CHANGES_REQUESTED** [review]
Thanks! Cool icons )

@vng do we need to fix something else?

File: data/categories.txt (hunk 3 above)
Comment (vng) [reply data/categories.txt:6991]: Well, yes, I'd merge them in categories like:
sport-9pin|sport-10pin


And in types_strings like:
```
[type.sport.9pin]
  ref = type.sport.10pin
```

File: data/visibility.txt [hunk 4]
```diff
@@ -970,10 +974,13 @@ world  00000000000000000000  +
         multi  00000000000000000000  -
         scuba_diving  00000000000000000000  -
         shooting  00000000000000000000  -
+        skateboard  00000000000000000000  -
```
Comment (biodranik) [inline data/visibility.txt:977]: @vng do we need this file if it's completely ignored now?

File: data/categories.txt [hunk 5]
```diff
@@ -6801,6 +6950,80 @@ vi:Quần vợt
...
+sw:Voliboli
+th:วอลเลย์บอล
+tr:Voleybol
+uk:Волейбол
+vi:Bóng chuyền
+zh-Hans:排球
+zh-Hant:排球
+
+sport-9pin|sport-10pin
+en:Nine-pin Bowling|Bowling
```
Comment (biodranik) [inline data/categories.txt:6991]: Let's make it simple "Bowling", I doubt that someone will use Nine-pin in the search.

File: data/visibility.txt (hunk 4 above)
Comment (vng) [reply data/visibility.txt:977]: No we don't.

Comment (vng) [reply data/visibility.txt:977]: @j13m126 No need any actions from your side.

**Comment by vng** [comment]
From CI:
```
Running style_symbols_consistency_test.cpp::Test_SymbolsConsistency
Test with map style MapStyleClear
Symbols mismatch: style MapStyleClear , density mdpi , missed [1: bowls-m ]
Symbols mismatch: style MapStyleClear , density hdpi , missed [1: bowls-m ]
Symbols mismatch: style MapStyleClear , density xhdpi , missed [1: bowls-m ]
Symbols mismatch: style MapStyleClear , density xxhdpi , missed [1: bowls-m ]
Symbols mismatch: style MapStyleClear , density xxxhdpi , missed [1: bowls-m ]
Symbols mismatch: style MapStyleClear , density 6plus , missed [1: bowls-m ]
Test with map style MapStyleDark
Symbols mismatch: style MapStyleDark , density mdpi , missed [1: bowls-m ]
Symbols mismatch: style MapStyleDark , density hdpi , missed [1: bowls-m ]
Symbols mismatch: style MapStyleDark , density xhdpi , missed [1: bowls-m ]
Symbols mismatch: style MapStyleDark , density xxhdpi , missed [1: bowls-m ]
Symbols mismatch: style MapStyleDark , density xxxhdpi , missed [1: bowls-m ]
Symbols mismatch: style MapStyleDark , density 6plus , missed [1: bowls-m ]
Test with map style MapStyleVehicleClear
Test with map style MapStyleVehicleDark
FAILED
style_tests/style_symbols_consistency_test.cpp:97 TEST(res) 
```

bowls-m.svg is deleted, but still present somewhere in mapcss files. Probably in styles-night or vehicle.

**Comment by vng** [comment]
bowls-m.svg still present in Icons.mspcss

**Comment by vng** [comment]
@j13m126 Please, check Icons.mapcss once again. Tests show that deleted bowls-m.svg still present in mapcss styles.

**Comment by endim8** [comment]
Thank you for the PR, sport icons definitely needed work :)

I've found a bug, table tennis tables use the generic sports ground icon when mapped as an area:
Screenshot_20220728-091237_Organic Maps

on osm: https://www.openstreetmap.org/way/975293363

**Comment by vng** [comment]
Great!
- Should fix existing outdated type like leisure-sports_centre-climbing, I can do it by myself on your branch.
- Lets find yoga.svg icon as well, so we can replace all outdated leisure-sports_centre-* with the new approach (sport=*). Because sports exist not only in sports_centre :)

**Comment by vng** [comment]
yoga and consistent gym looks good to me

**Comment by biodranik** [comment]
> For consistency, I removed the background circle for gym-m.svg (see screenshot above) as well, as fitness_stations are often close to sport pitches. gym-m.svg is used for leisure=fitness_station and leisure=fitness_centre.

fitness_centre is a business, and our general pattern is to use a circled icon for all businesses. It is not critical but may bring more consistency to the map. What do you think? Would it be possible to have two variants of the icon for these two different cases?

**Comment by vng** [comment]
I don't think that we need 2 versions of sports icons. drinking_water also may be a business, but we don't use circle there.

**Comment by vng** [comment]
Let's focus here on sport first. And update/merge PR, its a long story already ..

**Comment by vng** [comment]
If we are agree here, I can make final changes to update existing tags and tests.

**Comment by biodranik** [comment]
Is there a way to programmatically draw the icon's circle/square, change its color, or do whatever modification we need?

I'm ok to merge all sports first without a circle, but please create a separate issue for it so we can track and improve it later.

**Comment by vng** [comment]
There is, and we draw some custom user marks like this (speed cams), but for model POIs bette to have static icons.

**Comment by vng** [comment]
Final changes. PTAL.

File: data/strings/types_strings.txt [hunk 6]
```diff
@@ -11701,75 +11701,7 @@
...
-    sk = Strelnica
-    sv = Skjutbana
-    th = สนามยิงปืน
-    tr = Poligon
-    uk = Тир
-    vi = Nơi bắn súng
-    zh-Hans = 射击场
-    zh-Hant = 射擊場
-
-  [type.leisure.sports_centre.swimming]
```
Comment (biodranik) [inline data/strings/types_strings.txt:11738]: Should these translations go to type.sport.swimming?

File: data/strings/types_strings.txt [hunk 7]
```diff
@@ -11701,75 +11701,7 @@
     zh-Hans = 攀登中心
     zh-Hant = 攀登中心
 
-  [type.leisure.sports_centre.shooting]
```
Comment (biodranik) [inline data/strings/types_strings.txt:11704]: Should these translations go to type.sport.shooting?

File: map/style_tests/classificator_tests.cpp [hunk 8]
```diff
@@ -255,5 +255,34 @@ UNIT_TEST(Classificator_PoiPriority)
       {"railway", "station"}, {"railway", "halt"}, {"railway", "tram_stop"},
     }, {2, 5}, drule::symbol);
   }
+
+  /// @todo Check that all of sport=* icons priority is bigger than all of pitch, sport_center, playground.
```
Comment (biodranik) [inline map/style_tests/classificator_tests.cpp:259]: Classificator::ForEachInSubtree ?

File: data/strings/types_strings.txt (hunk 6 above)
Comment (vng) [reply data/strings/types_strings.txt:11738]: уже есть

File: data/strings/types_strings.txt (hunk 7 above)
Comment (vng) [reply data/strings/types_strings.txt:11704]: already exists

File: map/style_tests/classificator_tests.cpp (hunk 8 above)
Comment (vng) [reply map/style_tests/classificator_tests.cpp:259]: more tricky here, someday in next PR

**Review by biodranik — APPROVED** [review]

**Pull request merged**